### ISSUE TITLE
The next version of WC Vendors

### ISSUE THREAD

**Issue by dinhtungdu** [wcvendors/wcvendors#552]
This issue is a roadmap/checklist for our next version of WC Vendors which introduces a better file structure and naming inspired by WooCommerce, a new commission system, improved order management system, REST API and much more...

## Migration
- [ ] Move to the new file structure.
- [ ] Keep everything as 2.x version.
- [ ] Keep as many filters/actions as possible.
- [ ] Add notice/warning for deprecated actions/filter (`do_it_wrong` for example).
- [ ] Make use of CI and code quality tools.
- [ ] Add unit tests.

## New Order management system.
- [ ] Add Vendors order management screen for marketplace admin.
- [ ] Fully extend the WC_Order. Adding all necessary properties like line items, billing and shipping information, fee, tax, etc..
- [ ] Make use of vendor order instead of using old commission table
- [ ] Add tools to migrate/update order data for old versions.

## New Commission System.
- [ ] Create new WooCommerce data types to store commission data.
- [ ] Use a custom table to store data.
- [ ] Implementing line items for commission: admin commission, commission, shipping, fee, tax, etc..
- [ ] Add tools to convert old commission data to the new system.

## REST API
- [ ] Updating

*This version is in active development at [`next`](https://github.com/wcvendors/wcvendors/tree/next) branch.*

**Comment by digitalchild** [comment]
The migrate/update order data and existing sub orders should remove all unnecessary meta_data from the meta table.

**Comment by bporcelli** [comment]
Will the new commission system include an API for registering custom commission statuses? If not, is this something you would consider adding to your roadmap? Thanks!

**Comment by dinhtungdu** [comment]
Yeah, I think custom commission statuses should be implemented too.

**Comment by bporcelli** [comment]
Awesome! 👍

**Comment by digitalchild** [comment]
Move product import/export code from Pro to free so marketplace admins can import / export vendor products and assign them to the vendors.

**Comment by dinhtungdu** [comment]
Commission should include the commission setting at the creation time because commission for a product can be changed at any time. With the exact commission setting for a product, it's much easier to do refund and recalculate commission.
